tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/AsynchronousCSTest.java

```diff
@@ -144,205 +112,57 @@ public void testClassLevelAsyncIsNotFinished() {
-        Assert.assertEquals(executionRecord.toString(), "123", 
-                "The execution didn't happen in the expected order");
+        Task taskToPerform = new ServiceTask();
+        CompletionStage<Task> taskResult = client.serviceCS(taskToPerform).thenApply(v -> {
+            executionRecord.append("1");
+            return v;
+        }).thenApply(v -> {
+            executionRecord.append("2");
+            return v;
+        }).thenApply(v -> {
+            executionRecord.append("3");
+            return v;
+        });
 
-        try {
-            waitUntilCompleted(resultFuture);
-        }
-        catch (CompletionException e) {
-            handleCompletionException(e);
-        }
+        await().untilAsserted(() -> assertTrue(taskResult.toCompletableFuture().isDone()));
+        assertEquals(executionRecord.toString(), "123");
     }
 
     /**
-     * Test that the stage returned by calling an asynchronous method is 
+     * Test that the stage returned by calling an asynchronous method is
      * completed exceptionally if the method throws an exception
      */
     @Test
     public void testAsyncCompletesExceptionallyWhenExceptionThrown() {
-        CompletableFuture<Void> waitingFuture = newWaitingFuture();
-        
-        CompletionStage<Connection> resultFuture = client.serviceCS(waitingFuture, true);
-        waitingFuture.completeExceptionally(new SimulatedException("completedExceptionally"));
-
-        Assert.assertTrue(completesQuickly(resultFuture));
-        Assert.assertTrue(isCompletedExceptionally(resultFuture));
-        Assert.assertFalse(isCancelled(resultFuture));
-        assertThrowsExecutionExceptionWithCause(SimulatedException.class, CompletableFutureHelper.toCompletableFuture(resultFuture)::get);
-    }
-
-    /**
-     * Test that the stage returned by calling an asynchronous method is 
-     * completed exceptionally if the method returns a stage completed exceptionally
-     */
-    @Test
-    public void testAsyncCompletesExceptionallyWhenCompletedExceptionally() {
-        CompletableFuture<Void> waitingFuture = newWaitingFuture();
-        
-        CompletionStage<Connection> resultFuture = client.serviceCS(waitingFuture, false);
-        waitingFuture.completeExceptionally(new SimulatedException("completedExceptionally"));
-
-        Assert.assertTrue(completesQuickly(resultFuture));
-        Assert.assertTrue(isCompletedExceptionally(resultFuture));
-        Assert.assertFalse(isCancelled(resultFuture));
-        assertThrowsExecutionExceptionWithCause(SimulatedException.class, CompletableFutureHelper.toCompletableFuture(resultFuture)::get);
-    }
-
-    /**
-     * Ensure that any waiting futures get completed at the end of each test
-     * <p>
-     * Important in case tests end early due to an exception or failure.
-     */
-    @AfterTest
-    public void completeWaitingFutures() {
```

Comment:
This method was important to ensure that any waiting futures were cleaned up at the end of the test.

Ideally we should be using waitingFutures (or some other sort of synchronization) instead of sleeps as this allows test tasks to run only as long as their needed, rather than for a fixed length of time. In this case, I would expect to see some cleanup logic similar to this method to ensure that in the case of failure, we don't leave any threads waiting indefinitely for a signal from the test which will never arrive.